File: transaction-api-backend/src/load_data.js
const fs = require("fs");
const mysql = require("mysql");

// MySQL connection details
const connection = mysql.createConnection({
  host: "127.0.0.1",
  user: "root",
  password: "",
  database: "neowise",
});
const usersData = JSON.parse(fs.readFileSync('users.json', 'utf-8'));

const transactionsData = JSON.parse(fs.readFileSync('transactions.json', 'utf-8'));

// Connect to MySQL database
connection.connect((err) => {
    if (err) {
        console.error('Error connecting to MySQL:', err);
        return;
    }
    console.log('Connected to MySQL database.');
});

// Insert data into users table
usersData.forEach(user => {
    const { id, name, balance } = user;
    const query = `INSERT INTO users (userid, name, balance) VALUES (?, ?, ?)`;
    connection.query(query, [id, name, balance], (err, result) => {
        if (err) {
            console.error('Error inserting data into users table:', err);
            return;
        }
        console.log(`Inserted user ${id} ${name} into users table.`);
    });
});

// Insert data into transactions table
transactionsData.forEach(transaction => {
    const { transactionId, senderId, receiverId, amount, details } = transaction;
    const query = `INSERT INTO transactions (transactionId, senderId, receiverId, amount, details) VALUES (?, ?, ?, ?, ?)`;
    connection.query(query, [transactionId, senderId, receiverId, amount, details], (err, result) => {
        if (err) {
            console.error('Error inserting data into transactions table:', err);
            return;
        }
        console.log(`Inserted transaction ${transactionId} into transactions table.`);
    });
});

// Close MySQL connection
connection.end((err) => {
    if (err) {
        console.error('Error closing MySQL connection:', err);
        return;
    }
    console.log('MySQL connection closed.');
});

File: flask-backend/load_data.py
import json
import mysql.connector

# MySQL connection details
connection = mysql.connector.connect(
    host="127.0.0.1",
    user="root",
    password="",
    database="neowise"
)

# Function to read JSON data from file
def read_json(filename):
    with open(filename, 'r') as file:
        data = json.load(file)
    return data

# Insert data into users table
def insert_users(users_data):
    cursor = connection.cursor()
    for user in users_data:
        user_id = user['id']
        name = user['name']
        balance = user['balance']
        query = "INSERT INTO users (userid, name, balance) VALUES (%s, %s, %s)"
        cursor.execute(query, (user_id, name, balance))
        print(f"Inserted user {user_id} {name} into users table.")
    connection.commit()
    cursor.close()

# Insert data into transactions table
def insert_transactions(transactions_data):
    cursor = connection.cursor()
    for transaction in transactions_data:
        transactionId = transaction['transactionId']
        senderId = transaction['senderId']
        receiverId = transaction['receiverId']
        amount = transaction['amount']
        details = transaction['details']
        query = "INSERT INTO transactions (transactionId, senderId, receiverId, amount, details) VALUES (%s, %s, %s, %s, %s)"
        cursor.execute(query, (transactionId, senderId, receiverId, amount, details))
        print(f"Inserted transaction {transactionId} into transactions table.")
    connection.commit()
    cursor.close()

# Read users data from JSON file
users_data = read_json('users.json')

# Read transactions data from JSON file
transactions_data = read_json('transactions.json')

# Insert users data into users table
insert_users(users_data)

# Insert transactions data into transactions table
insert_transactions(transactions_data)

# Close MySQL connection
connection.close()
